sqlx-macros-core/src/query/mod.rs

```diff
@@ -182,7 +165,7 @@ pub fn expand_input<'a>(
 
             // Check SQLX_OFFLINE_DIR, then local .sqlx, then workspace .sqlx.
             let dirs = [
-                |_: &Metadata| env("SQLX_OFFLINE_DIR").ok().map(PathBuf::from),
```

Comment:
This line would have made 0.8.5 work.  Now that it is not checked, I believe this causes #3961

Reply:
I believe you're correct. We only read it from the `.env` but not the actual process environment: https://github.com/launchbadge/sqlx/blob/24317d5eab40fbc33caf1142946e2f39caad73ea/sqlx-macros-core/src/query/mod.rs#L127

We only actually check the process environment when we go to write the data out: https://github.com/launchbadge/sqlx/blob/24317d5eab40fbc33caf1142946e2f39caad73ea/sqlx-macros-core/src/query/mod.rs#L386

However, while we should fix this for existing users, I think the offline directory should be configurable in the `sqlx.toml` (#3383).